kube/objects.go

```diff
@@ -159,7 +193,79 @@ func (c *Client) FetchObjects(ctx context.Context, filter ObjectFilter) (*Object
 		return nil, err
 	}
 
-	return objects, nil
+	return objectsWithoutNils(objects), nil
+}
+
+func annotateFetchError(kind string, err error) error {
+	if err == nil {
+		return nil
+	}
+	if kerrors.IsNotFound(err) {
+		// Resource doesn't exist in this cluster's version, so there aren't any
+		// objects to list and check.
+		return nil
+	}
+
+	return fmt.Errorf("failed to fetch %s: %s", kind, err)
+}
+
+func objectsWithoutNils(objects *Objects) *Objects {
+	if objects.Nodes == nil {
```

Comment:
If I delete the content of this function except for the return statement, the tests still pass.

We should probably add a test to validate that we are handling not-found resources appropriately.